### PR TITLE
Point openverse frontend to api.openverse.org

### LINKED ISSUE #3741: Point frontend to `api.openverse.org`
## Description

Part of https://docs.openverse.org/projects/proposals/infrastructure_improvements/20240105_implementation_plan_move_remaining_services_to_openverse_org.html#move-the-django-api-to-api-openverse-org

Blocked by https://github.com/WordPress/openverse-infrastructure/issues/778

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #3741 by @sarayourfriend

## Description

Update references to the Openverse API in the frontend to use api.openverse.org, including updating test tapes and fixtures.

## Testing Instructions

CI should pass for all frontend tests, both unit and e2e. Run the frontend app locally using `just p frontend dev` and confirm it uses `api.openverse.org` for API requests by checking requests in your browser's developer console.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/test/proxy.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const talkback = require("talkback")
 /** @typedef {ReturnType<TalkbackOptions['tapeDecorator']>} Tape */
 
 const port = 49153
-const host = "https://api.openverse.engineering"
+const host = "https://api.openverse.org"
```
Comment (obulat) [inline frontend/test/proxy.js:26]: I wonder if this should use the env variable...

**Review by obulat — APPROVED** [review]
Exciting! ✨

**Review by dhruvkb — APPROVED** [review]
I'm not reviewing the `tapes/` but the rest LGTM 🚀!

File: frontend/test/proxy.js (hunk 1 above)
Comment (sarayourfriend) [reply frontend/test/proxy.js:26]: Could do! Good for another issue, I think though.

**Pull request merged**